### PR TITLE
Cleanup GregorianCalendar tests

### PR DESCRIPTION
**Pull request by hughbe**
- Use xunit
- Refactor out some common code and utility methods
- Fix some duplicate tests that should have been testing other things.
- Contributes to #2336

### CODE REVIEW

**Comment by stephentoub** [comment]
Test Innerloop Ubuntu Debug Build and Test please
Test Innerloop Ubuntu Release Build and Test please

**Comment by stephentoub** [comment]
cc: @tarekgh

File: src/System.Globalization.Calendars/tests/GregorianCalendar/GregorianCalendarTests.Utilities.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            int maxYear = s_calendar.MaxSupportedDateTime.Year;
+            int minYear = s_calendar.MinSupportedDateTime.Year;
+            return minYear + s_randomDataGenerator.GetInt32(-55) % (maxYear + 1 - minYear);
+        }
+
+        public static int RandomLeapYear()
+        {
+            // A leap year is any year divisible by 4 except for centennial years(those ending in 00)
+            // which are only leap years if they are divisible by 400.
+            int year = ~(~RandomYear() | 0x3); // retVal will be divisible by 4 since the 2 least significant bits will be 0
```
Comment (tarekgh) [inline src/System.Globalization.Calendars/tests/GregorianCalendar/GregorianCalendarTests.Utilities.cs]: please change the comments as we are not using retVal name anymore

**Comment by tarekgh** [comment]
I added very minor comment, LGTM otherwise

File: src/System.Globalization.Calendars/tests/GregorianCalendar/GregorianCalendarTests.Utilities.cs (hunk 1 above)
Comment (hughbe) [inline src/System.Globalization.Calendars/tests/GregorianCalendar/GregorianCalendarTests.Utilities.cs]: good catch, fixed

**Comment by tarekgh** [comment]
it will be good if you squash the 2 commits into one.

**Comment by hughbe** [comment]
Sure thing boss!

**Comment by tarekgh** [comment]
thanks @hughbe :smiley:

**Comment by hughbe** [comment]
Rebased to fix a merge conflict

**Pull request merged**